Khan/perseus
# Implement the "sides" snapTo type for Polygon

## Summary:

Polygon previously was only able to snap to grid lines and angles. Functionality was needed to allow it to snap to whole side lengths regardless of the grid, snapStep, or angles. This implements that feature.

Known Strange Behavior: 
In the image below, the gallery shows it is possible for the lines to end up on top of each other when snapping to lines (shown on the right). When snapping to grid (on the left), this doesn't happen. 
![image](https://github.com/Khan/perseus/assets/60857422/c8808f93-bb43-4164-a674-032fe980913d)

This behavior is not a regression as it also happens on legacy graphs.
![image](https://github.com/Khan/perseus/assets/60857422/202f8146-734c-4f11-9340-3360c0e14ed5)



Issue: LEMS-1902

## Test plan:
- Confirm all tests pass
- Confirm test coverage for new code is good
- Run `yarn dev` in the command line
- Navigate to localhost:5173
- Turn on mafs flags for all graphs
- Confirm the new gallery test for snapTo sides works correctly
- Confirm the other types of graphs work correctly
- Go to the flipbook and enter the following content for polygon:
```
{"content":"[[☃ interactive-graph 1]]","images":{},"widgets":{"interactive-graph 1":{"alignment":"default","graded":true,"options":{"backgroundImage":{"url":""},"correct":{"coords":[[-2,1],[-2,6],[-5,6],[-5,1]],"numSides":4,"snapTo":"sides","type":"polygon"},"graph":{"numSides":7, "showAngles":true, "showSides": true, "snapTo":"grid","type":"polygon"},"gridStep":[1,1],"labels":["x","y"],"markings":"graph","range":[[-10,10],[-10,10]],"rulerLabel":"","rulerTicks":10,"showProtractor":false,"showRuler":false,"showTooltips":false,"snapStep":[2,2],"step":[2,2]},"static":false,"type":"interactive-graph","version":{"major":0,"minor":0}}}}
```
- Move the points in "snapTo" grid mode and confirm the points snap to the grid
- Move the polygon and confirm it snaps to the grid
- Change the "snapTo" property of the graph object in the content to `sides`
- Move the points and confirm the sides are whole numbers and do not snap to the grid or angles
- Move the entire polygon and confirm it does not snap to the grid or angles and the sides stay whole numbers
- Confirm no errors appear in the console
